examples/dtls-sock/dtls-client.c

```diff
@@ -68,18 +76,102 @@ static const credman_credential_t credential = {
 };
 #endif
 
-static int client_send(char *addr_str, char *data, size_t datalen)
+static void _timeout_handler(event_t *);
+
+static uint8_t _recv_buf[512];
+static sock_udp_t _udp_sock;
+static sock_dtls_t _dtls_sock;
```

Comment:
Why are those global?

Reply:
Ah `event_timeout_set()` doesn't accept optional payload :\

Reply:
Also (and much more the reason why): the function initiating the hand-shake finishes after sending the hello message, popping potentially stored socks there from stack ;-).

Reply:
Probably worth adding a comment about that

Reply:
This is C basics. If we need a comment for _that_ we should add it to any global variable.

Reply:
Not using global variables when there might be multiple instances of a thing could also be considered C basics.

> the function initiating the hand-shake finishes after sending the hello message, popping potentially stored socks there from stack

Is certainly not included in my copy of K & R :wink: 

See, my credo is if something pops up with needing clarification during a code review, we better note it down. So if someone who knows as little about the code as me will read it in the future, they will not have to hunt you down to understand certain design decisions.

Reply:
If your K&R does not have the fact in it, that local variables should not referenced after you left the current scope something is seriously wrong with it ;-). But fine, let me try to cook something up :-).

Reply:
Arg sorry this ist just the example - so yea that makes sense.
I was somehow under the assumption that  this was inside `pkg/tinydtls` so the global socket raised an eyebrow.